File: AcademicWebApp/src/main/java/com/example/AcademicWebApp/Models/Grade.java
package com.example.AcademicWebApp.Models;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity(name = "grade")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@IdClass(GradeId.class)
public class Grade implements Serializable {


    @Column(name = "gradevalue")
    private Integer gradevalue;

    @Id
    @Column(name = "username")
    private String username;

    @Id
    @Column(name = "cid")
    private Integer cid;

    /*
    @Override
    public String toString() {
        return this.username + ", " + this.name + ", " + this.group1 + ", " + this.group2;
    }
    */

}

File: DataGenerator/domain/grade.py


class Grade:

    def __init__(self, value, username, cid):
        self.__value = value
        self.__username = username
        self.__cid = cid

    @property
    def value(self):
        return self.__value

    @property
    def username(self):
        return self.__username

    @property
    def cid(self):
        return self.__cid

    def __str__(self):
        return str(self.__value) + ", '" + self.__username + "', " + str(self.__cid)
